internal/bucket/lifecycle/lifecycle.go

```diff
@@ -261,7 +261,7 @@ func (lc Lifecycle) FilterRules(obj ObjectOpts) []Rule {
 		if !strings.HasPrefix(obj.Name, rule.GetPrefix()) {
 			continue
 		}
-		if !rule.Filter.TestTags(obj.UserTags) {
+		if !obj.DeleteMarker && !rule.Filter.TestTags(obj.UserTags) {
```

Comment:
@harshavardhana maybe for now, we can check if the current rule contains ExpireAllObjects; I am looking what is the behavior of ILM in this case (I am guess this code is correct after all)